test/test_repo.py

```diff
@@ -1377,13 +1377,23 @@ def test_rebasing(self, rw_dir):
 
     @with_rw_directory
     def test_do_not_strip_newline_in_stdout(self, rw_dir):
+        r = self.create_repo_commit_hello_newline(rw_dir)
+        self.assertEqual(r.git.show("HEAD:hello.txt", strip_newline_in_stdout=False), "hello\n")
+
+    def create_repo_commit_hello_newline(self, rw_dir):
         r = Repo.init(rw_dir)
         fp = osp.join(rw_dir, "hello.txt")
         with open(fp, "w") as fs:
             fs.write("hello\n")
         r.git.add(Git.polish_url(fp))
         r.git.commit(message="init")
-        self.assertEqual(r.git.show("HEAD:hello.txt", strip_newline_in_stdout=False), "hello\n")
+        return r
+
+    @with_rw_directory
+    def test_warn_when_strip_newline_in_stdout(self, rw_dir):
```

Comment:
Let's put the additional assertion done in `test_do_not_strip_newline_in_stdout` here so that it's clearer what's under test: The deprecation warning if `git show` is used and `strip_newline_in_stdout=True`. This way, `test_do_not_strip_newline_in_stdout` can be removed. I'd also hope that the assertion then can be changed to use the default value for `strip_newline_in_stdout`, which is probably `True`, the cause of the confusion in this case.